### ISSUE TITLE
Bulk import/create users

### ISSUE THREAD

**Issue by iAugur** [kendraio/kendra_home#23]
Add the ability to bulk create users on Kendra_home
Review available modules e.g. https://www.drupal.org/project/user_import

**Comment by BBGuy** [comment]
Created a new branch to test https://github.com/kendraio/kendra_home/tree/dev_27

**Comment by BBGuy** [comment]
*To deploy*
* drush en user_import
* Add "import users" permission to admin
* set privet file system pass /admin/config/media/file-system
* Create new import: admin/people/user_import/errors/6

**Comment by BBGuy** [comment]
Tested the module on the test brunch and:

**The good**
+ only new users are created (matched by user name) 
+ option to update users
+ user name can be set from a CSV username column or by combining CSV columns like First name + last name
+ supports most of our fields in both user and user profile
+ is extendable through an API

**The bad**
- Field Website is not supported
- Field Full name is not supported 

**Next stage**
@dahacouk - if you can provide some data we can run a test - assigning to you and once you have data assign back to me

**Comment by dahacouk** [comment]
That's a shame that `full name` is not supported. All we have is email and full name. Anything we can do about this?

**Comment by BBGuy** [comment]
yes probably it doesn't do it out of the box but claims to be extensible. will investigate.

**Comment by BBGuy** [comment]
I managed to add support for the safeword field and full name is now importing, however the URL is not set from the safe value. Will investigate.

**Comment by BBGuy** [comment]
OK this my understanding about how the URL is set
1) set the full name field - Sets the safeword human
2) Safeword module sets the machine name from the human name
3) The real name module uses the full name field
4) The *user:name-raw* token in the URL pattern "users/[user:name-raw]" point at the real name.

user:name-raw: The raw login name of the user account, unprocessed by the RealName module.

so looks like real name is not triggering correctly or the data is not available to it when it does.

**Comment by BBGuy** [comment]
So looks like the machine name is not automatically set on entity update and is controlled by the UI. 
Looking at the Migrate implementation the machine name is set directly not derived from the name.
so we have two options:
1) include a url-safe value to the field. 
2) derive one in code.

**Comment by BBGuy** [comment]
Updated code with @iAugur function to set the machine name.
Import now working.

**Comment by BBGuy** [comment]
All code is now live. Still to do -
-[ ] enable module
-[ ] create sample import and test on dev site
-[ ] Get file format from @dahacouk  and create import

**Comment by BBGuy** [comment]
import is now setup on live and using the flowing format
```
"user","password","email","full name","company","Job title"
"guy test 1","pita1","guytest1@blue-bag.com","Guy Schneerson test","blue-bag ltd","Drupl developer"
"guy test 2","pita2","guytest2@blue-bag.com","Guy Testerson","Range of vision","Musician"
```
@dahacouk - Can you review the fields in the above sample CSV snippet.

**Comment by dahacouk** [comment]
I guess we shouldn't have "company","Job title" fields.

**Comment by BBGuy** [comment]
Removed "company" & "Job title" from import

**Comment by BBGuy** [comment]
Tested and works with core update to 7.35 #37

**Comment by BBGuy** [comment]
one small issue with the import is we are getting a warning:
Warning: Creating default object from empty value in _user_import_settings_save() (line 574 of /var/www/kendra/kendra-home/dev/htdocs/sites/all/modules/contrib/user_import/user_import.module).
See issue https://www.drupal.org/node/2199415
However this is not visible to users and the import works so will not bother with for now.

**Comment by BBGuy** [comment]
Created two versions of the import 
1) Activates the account and does not send email to user
2) sends a welcome email with a one time login link and user need to login and set their password before the account is activated.

**Comment by BBGuy** [comment]
The email format I used is:
Welcome to Kendra hub

To activate your account use the following one time login url:  !login_url

This link can only be used once to log in and will lead you to a page where you can set your password.

After setting your password, you will be able to log in at !login_uri in the future using:

username: !username
password: Your password

Kind regards,
The Kendra team

**Comment by dahacouk** [comment]
When Drupal sends out the email it is hard wrapped. I'd like it to be unwrapped.

**Comment by BBGuy** [comment]
tested both user imports on live and both worked so only issue is the formatting.
- [ ]@dahacouk - let me know when you finished with your imported test account and ill delete it.

**Comment by dahacouk** [comment]
- [ ] I want to check what happens when I import a new user with a new email but the same username.

**Comment by dahacouk** [comment]
@BBGuy I'm getting the error "Field Website is not supported" when I start a bulk user import – the 2 screen just after specifying the CSV file.

**Comment by dahacouk** [comment]
@BBGuy Errors don't seem to be viewable. Should they be? See:
* https://www.kendra.io/admin/people/user_import
* https://www.kendra.io/admin/people/user_import/errors/19

**Comment by dahacouk** [comment]
When importing need to set *Appear in listings and project pages* to off. Need to check if this is something that can be done with user setting.

**Comment by dahacouk** [comment]
@BBGuy we need to set *Appear in listings and project pages* to off when we do an import.

**Comment by iAugur** [comment]
Patched VBO os the modify was causing an error for users:
https://www.drupal.org/node/2305227

**Comment by dahacouk** [comment]
I'm assigning this issue to me as I reckon it is all done and will close this week after I have more time to review.
